airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -696,10 +698,11 @@ def cleanup(self, pod: k8s.V1Pod, remote_pod: k8s.V1Pod):
                         f"Pod {pod and pod.metadata.name} returned exit code "
                         f"{self.skip_on_exit_code}. Skipping."
                     )
-            raise AirflowException(
-                f"Pod {pod and pod.metadata.name} returned a failure:\n{error_message}\n"
-                f"remote_pod: {remote_pod}"
-            )
+            raise AirflowException('\n'.join(filter(None, [
+                f"Pod {pod and pod.metadata.name} returned a failure.",
+                error_message,
+                f"remote_pod: {remote_pod}" if self.log_pod_on_failure else None
```

Comment:
```suggestion
                f"remote_pod: {remote_pod}" if self.log_pod_spec_on_failure else None
```

Reply:
yeah, I've already fixed that. My bad while parameter renaming.